### DISCUSSION TITLE
UI responsivness with locks - Intermidiate state?

### DISCUSSION THREAD

**Discussion by RAYs3T** [SukramJ/aiohomematic#806] [Ideas]
Is there any idea or work-a-round for delays when locking/unlocking a homematic Lock?

If you click on unlock for example, it takes between 500ms and 3 seconds before it shows "unlocking" as state.
I assume this is because the hahomematic only displays the real state (as soon as the lock really moves).

But this gives the user the feel that "nothing is happening" - hence, pressing the button again, triggering two events.

Maybe some intermediate "requesting xxx" state is a good idea? Or would that be something to implement entirely in the UI?

See [this video](https://nc.edging.tech/s/E2yipreEGCMtSpT) for better understanding.

**Comment by SukramJ** [comment]
We report is_open, is_closed, is_locking and is is_unlocking to the frontend based on states of the device, and this is what HA [supports](https://developers.home-assistant.io/docs/core/entity/lock#properties).
There are not further intermediate states, that could be used in HA, or that are supported by a homematic lock.
I think the delay is related to the wireless communication.

**Reply by RAYs3T** [reply]
The delay is caused on the route behind the CCU for sure. The command is executed directly on the press of the button, but takes longer.

I'm just looking for a solution to get the UI more fluent.
But with the lock class only supporting these states I see why this is not easily possible.

Maybe I have to create a wrapper for that lock in HA instead which has more "states". Just so it does not look like it's doing nothing.
